### DISCUSSION TITLE
Enabling and disabling

### DISCUSSION THREAD

**Discussion by Longhorn86** [zauberzeug/nicegui#560] [Q&A] [answered]
Hi, and thank you for this project, it is very interesting!

There has been a couple of questions regarding enabling/disabling things, like buttons (e.g. #293 and #480). The answers in the previous issues refers to using .props('disabled')

If I do this (drag the slider around):
```py
s = ui.slider(min=-100, max=100, value=0).props('disabled')
ui.label().bind_text_from(s, 'value')
```
I would expect the grayed out slider to be 'fixed' in place and the value not to change in such situation.

Also if I do
```py
ui.number(label='Value 1', value=10, format='%.1f')
ui.number(label='Value 2', value=1, format='%.1f').props('disabled')
```
the fact that the Value 2 -field is disabled, is not at all visible in the UI.

I'd expect the UI element to be un-alterable by the user when the element is disabled and this should work for all relevant element types, most (if not all) basic elements for example.

Is there a way to achieve the 'grayed out' behavior that I'm describing currently?

**Comment by falkoschindler** [comment, marked as answer]
Oh, very interesting observation, @Help-i-am-lost!

It looks like the prop `disabled` is an HTML attribute that works _kind of_. But the correct way would be to use the Quasar prop `disable` (without "d"). Then your code examples work as expected.

I'll update the two discussions you're referencing.

**Reply by Longhorn86** [reply]
Whoa.. Okay, I did not expect that kind of a reason. But now it works as I thought it should. Thank you! :+1:

**Reply by fdinhSG** [reply]
Hey! I think this is a pretty useful feature that's not clearly defined in the documentation. It might be worth mentioning it!

**Reply by falkoschindler** [reply]
I just created PR https://github.com/zauberzeug/nicegui/pull/797 proposing an official API for enabling and disabling input elements.
